cmake/maintainer.cmake

```diff
@@ -29,11 +28,20 @@ SET(MY_WARNING_FLAGS
   -Wno-null-conversion
   -Wno-unused-parameter
   -Wno-unused-private-field
-  -Woverloaded-virtual
-  -Wnon-virtual-dtor
   -Wvla
   -Wwrite-strings
   -Werror
+)
+
+# Common warning flags for GCC and Clang
+SET(MY_CC_WARNING_FLAGS
+  -Wdeclaration-after-statement
```

Comment:
Should not that be "MY_C_WARNING_FLAGS"?

Reply:
I changed that to MY_CC_WARNING_FLAGS so it's consistent with my calling of the C specific functions something with CC, instead of the CXX for C++.

Reply:
I see, but it needs to be C , not CC  to be consistent with the rest of CMake naming conventions :)